tests/Unit/Helpers/DataStructures/Tensor/Expressions/EvaluateRank2TestHelpers.hpp

```diff
@@ -0,0 +1,177 @@
+          typename RhsTensorIndexTypeList, typename TensorIndexA,
+          typename TensorIndexB>
+void test_evaluate_rank_2_core(
+    const TensorIndexA& tensorindex_a, const TensorIndexB& tensorindex_b) {
+  Tensor<Datatype, RhsSymmetry, RhsTensorIndexTypeList> R_ab{};
+  std::iota(R_ab.begin(), R_ab.end(), 0.0);
+
+  const size_t dim_a = tmpl::at_c<RhsTensorIndexTypeList, 0>::dim;
+  const size_t dim_b = tmpl::at_c<RhsTensorIndexTypeList, 1>::dim;
+
+  // L_{ab} = R_{ab}
+  const auto L_ab = TensorExpressions::evaluate<TensorIndexA, TensorIndexB>(
+      R_ab(tensorindex_a, tensorindex_b));
+  // L_{ba} = R_{ab}
+  const auto L_ba = TensorExpressions::evaluate<TensorIndexB, TensorIndexA>(
+      R_ab(tensorindex_a, tensorindex_b));
+
+  for (size_t i = 0; i < dim_a; ++i) {
+    for (size_t j = 0; j < dim_b; ++j) {
+      // For L_{ab} = R_{ab}, check that L_{ij} == R_{ij}
+      CHECK(L_ab.get(i, j) == R_ab.get(i, j));
+      // For L_{ba} = R_{ab}, check that L_{ji} == R_{ij}
+      CHECK(L_ba.get(j, i) == R_ab.get(i, j));
+    }
+  }
+}
+
+/// \ingroup TestingFrameworkGroup
+/// \brief Iterate testing of evaluating a single rank 2 Tensors on multiple
+/// Frame types and dimension combinations for nonsymmetric indices
+///
+/// \details `TensorIndexA` and `TensorIndexB` can be any type of TensorIndex
+/// and are not necessarily `ti_a_t` and `ti_b_t`. The "A" and "B" suffixes just
+/// denote the ordering of the generic indices of the RHS tensor expression. In
+/// the RHS tensor expression, it means `TensorIndexA` is the first index used
+/// and `TensorIndexB` is the second index used.
+///
+/// \tparam Datatype the type of data being stored in the Tensors
+/// \tparam TensorIndexTypeA the \ref SpacetimeIndex "TensorIndexType" of the
+/// first index of the RHS Tensor
+/// \tparam TensorIndexTypeB the \ref SpacetimeIndex "TensorIndexType" of the
+/// second index of the RHS Tensor
+/// \tparam TensorIndexA the type of the first TensorIndex used on the RHS of
+/// the TensorExpression, e.g. `ti_a_t`
+/// \tparam TensorIndexB the type of the second TensorIndex used on the RHS of
+/// the TensorExpression, e.g. `ti_B_t`
+/// \tparam ValenceA the valence of the first index used on the RHS of the
+/// TensorExpression
+/// \tparam ValenceB the valence of the second index used on the RHS of the
+/// TensorExpression
+/// \param tensorindex_a the first TensorIndex used on the RHS of the
+/// TensorExpression, e.g. `ti_a`
+/// \param tensorindex_b the second TensorIndex used on the RHS of the
+/// TensorExpression, e.g. `ti_B`
+template <typename Datatype,
+          template <size_t, UpLo, typename> class TensorIndexTypeA,
+          template <size_t, UpLo, typename> class TensorIndexTypeB,
+          UpLo ValenceA, UpLo ValenceB, typename TensorIndexA,
+          typename TensorIndexB>
+void test_evaluate_rank_2_no_symmetry(const TensorIndexA& tensorindex_a,
+                                      const TensorIndexB& tensorindex_b) {
```

Comment:
@nilsdeppe I interpreted your other review comment to mean to add `noexcept` to all of the `test_evaluate_rank_#_core` and `test_evaluate_rank_0` functions (i.e. all the innermost functions that use `CHECK`), but do you also want `noexcept` used with *all* of the functions defined in these `EvaluateRank#TestHelpers.hpp` files, like these ones?

Reply:
Yep! Our general policy is "noexcept every function". The only exception (no pun intended) is the option parsing code